### PR TITLE
drivers: pcie: add config support for enable PRT PCI Routing Table

### LINKED ISSUE #63504: intel_ehl_crb board does not work with ACPI IRQ allocation
Interrupt allocation does not work for `intel_ehl_crb` board. The following code is not working:
https://github.com/zephyrproject-rtos/zephyr/blob/22f09e9fa03fcf88fcc0832857a4d77f8ea0db8c/drivers/pcie/host/pcie.c#L293-L299

since `CONFIG_ACPI` is always enabled. If I remove all this and leave only `irq = arch_irq_allocate();` then interrupt allocation works again.

### PR DESCRIPTION
**Pull request by najumon1980**
Add config support for enable IRQ information retrieve via ACPI PRT (PCI Routing Table). 

Fixes #63504

### CODE REVIEW

**Comment by najumon1980** [comment]
Waiting for Andrei to confirm

**Comment by finikorg** [comment]
> Waiting for Andrei to confirm

This works, but the question is: do we need ACPI for those boards? Was ACPI only used for IRQ allocation?

**Comment by najumon1980** [comment]
> > Waiting for Andrei to confirm
> 
> This works, but the question is: do we need ACPI for those boards? Was ACPI only used for IRQ allocation?

ACPI is not only use for retrieving interrupt routing info, but for CPU config etc. as well.

File: modules/acpica/Kconfig [hunk 1]
```diff
@@ -8,7 +8,7 @@ menu "ACPI driver support"
 
 config ACPI_DSDT_SUPPORT
 	bool  "Build source code for DSDT ACPICA support"
-	default y if PCIE
+	default y if PCIE && ACPI
```
Comment (jhedberg) [inline modules/acpica/Kconfig:11]: Shouldn't this have a `depends on ACPI`? If you have that then the original `if PCIE` should be enough

**Review by tbursztyka — DISMISSED** [review]

File: modules/acpica/Kconfig (hunk 1 above)
Comment (finikorg) [reply modules/acpica/Kconfig:11]: Should it be something like
```
config ACPI
	bool

if ACPI

menu "ACPI driver support"

.....

endif 
```

Comment (jhedberg) [reply modules/acpica/Kconfig:11]: @finikorg sure, that might be better, especially if we introduce more ACPI_* options in the future

Comment (najumon1980) [reply modules/acpica/Kconfig]: updated

**Review by jhedberg — DISMISSED** [review]
Looks good now, thanks!

**Comment by jhedberg** [comment]
@najumon1980 there are a couple of build failures (at least on `qemu_x86` as well as `acrn_ehl_crb`

**Comment by finikorg** [comment]
> @najumon1980 there are a couple of build failures (at least on `qemu_x86` as well as `acrn_ehl_crb`

Talking about deprecation, `acrn_ehl_crb` is a good candidate for it. I recently tested ACRN and used `acrn` board on EHL.

File: modules/acpica/Kconfig (hunk 1 above)
Comment (finikorg) [reply modules/acpica/Kconfig:13]: Maybe because `#` is a comment in Kconfig?

File: modules/acpica/Kconfig [hunk 2]
```diff
@@ -1,15 +1,19 @@
 # Copyright (c) 2023 Intel Corporation.
 # SPDX-License-Identifier: Apache-2.0
 
+menu "ACPI driver support"
+
 config ACPI
 	bool
 
-menu "ACPI driver support"
+#if ACPI
```
Comment (finikorg) [inline modules/acpica/Kconfig:9]: `if ACPI`

**Review by finikorg — CHANGES_REQUESTED** [review]
`#` is a comment in Kconfig

File: modules/acpica/Kconfig (hunk 1 above)
Comment (najumon1980) [reply modules/acpica/Kconfig]: > Maybe because `#` is a comment in Kconfig?

yes, you are right, just missed it!

**Comment by najumon1980** [comment]
> > @najumon1980 there are a couple of build failures (at least on `qemu_x86` as well as `acrn_ehl_crb`
> 
> Talking about deprecation, `acrn_ehl_crb` is a good candidate for it. I recently tested ACRN and used `acrn` board on EHL.

Yes we can remove the EHL ACRN since it does not enable any passthrough devices (PCI devices) and the board config is same as "acrn" board.

File: modules/acpica/Kconfig [hunk 3]
```diff
@@ -1,15 +1,19 @@
 # Copyright (c) 2023 Intel Corporation.
 # SPDX-License-Identifier: Apache-2.0
 
+menu "ACPI driver support"
```
Comment (finikorg) [inline modules/acpica/Kconfig:4]: should we put `menu` after `if` ?

File: modules/acpica/Kconfig (hunk 1 above)
Comment (finikorg) [reply modules/acpica/Kconfig]: @najumon1980 you put `menu` out of the `if`, why? Have you seen my sample above? Does it work?

**Review by finikorg — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by finikorg** [comment]
> > > @najumon1980 there are a couple of build failures (at least on `qemu_x86` as well as `acrn_ehl_crb`
> > 
> > 
> > Talking about deprecation, `acrn_ehl_crb` is a good candidate for it. I recently tested ACRN and used `acrn` board on EHL.
> 
> Yes we can remove the EHL ACRN since it does not enable any passthrough devices (PCI devices) and the board config is same as "acrn" board.

We can remove the `acrn_ehl_crb` in the next release, need to update ACRN documentation.

**Pull request merged**